src/interpreter/InterpretAllOf.ts

```diff
@@ -14,7 +14,13 @@ import { isModelObject } from './Utils';
  * @param interpreterOptions to control the interpret process
  */
 export default function interpretAllOf(schema: InterpreterSchemaType, model: CommonModel, interpreter : Interpreter, interpreterOptions: InterpreterOptions = Interpreter.defaultInterpreterOptions): void {
-  if (typeof schema === 'boolean' || schema.allOf === undefined) {return;}
+  if (
+    typeof schema === 'boolean' ||
+    schema.allOf === undefined ||
+    schema.oneOf
```

Comment:
Hmm, I see we have a bit of a problem with detecting when an interpreter should be run or not 🤔Especially if we find more patterns to look for...

While implementing this, did you think about a way we can improve this?

Reply:
Maybe the patterns should export something like `isPatternX` that then can be used as `!isPatternX()` to blacklist when interpretations should not happen 🤔 

Not saying we should do this now, just want to hear your opinion 🙂

Reply:
Hmm, good question. I can't think of any obvious ways to improve this right now. We need to check what schema properties exist in every interpreter anyway, so I'm not sure if `isPatternX` makes sense.

Reply:
What if the `interpretOneOfWithAllOf` exposed the following:

```ts
export isPatternOneOfWithAllOf = (....) => {
  if (
    typeof schema === 'boolean' ||
    !schema.oneOf ||
    !schema.allOf ||
    schema.properties ||
    interpreterOptions.allowInheritance
  ) { 
    return false;
  }
  return true;
}
```

and then the core allOf and oneOf would use it like the following: 

```ts
  if (
    typeof schema === 'boolean' ||
    schema.allOf === undefined ||
    isPatternOneOfWithAllOf(...)
```

we would of course have some duplicate checks but I think its worth it considering 🤔